src/mlpack/core/util/io.cpp

```diff
@@ -267,3 +267,75 @@ void IO::ClearSettings()
   GetSingleton().aliases = persistentAliases;
   GetSingleton().functionMap = persistentFunctions;
 }
+
+void IO::CheckInputMatrices()
+{
+  typedef typename std::tuple<data::DatasetInfo, arma::mat> TupleType;
+  std::map<std::string, util::ParamData>::iterator itr;
+
+  for (itr = IO::Parameters().begin(); itr != IO::Parameters().end(); ++itr)
+  {
+    std::string paramName = itr->first;
+    std::string paramType = itr->second.cppType;
+    std::string errMsg1 = "The input " + paramName + " has NaN values.";
+    std::string errMsg2 = "The input " + paramName + " has inf values.";
+
+    if (paramType == "arma::mat")
+    {
+      if (IO::GetParam<arma::Mat<double>>(paramName).has_nan())
+        Log::Fatal << errMsg1 << std::endl;
+
+      if (IO::GetParam<arma::Mat<double>>(paramName).has_inf())
```

Comment:
@rcurtin Do we have a string type mapping, similar to what's implemented in bandicoot? If we do I think we could simplify the block here.

Reply:
You're right, but I can't find one implemented for the Python bindings.  I don't know if it makes sense to add one just for this usage though.

Reply:
Yeah only if we had one already, which I wasn't sure I searched for the right thing.